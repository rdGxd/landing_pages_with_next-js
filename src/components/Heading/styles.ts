import styled, { DefaultTheme, css } from "styled-components";

type HeadingStyleProps = {
  $uppercase?: boolean;
  $colorDark?: boolean;
  size?: "big" | "medium" | "small" | "huge";
};

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? "uppercase" : "none"};
`;

export const Title = styled.h1<HeadingStyleProps>`
  ${({ theme, $colorDark, size, $uppercase }) => css`
    color: ${$colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase($uppercase)};
  `}
`;
