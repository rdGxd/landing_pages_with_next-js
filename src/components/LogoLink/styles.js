import Link from "next/link";
import styled from "styled-components";

export const Container = styled(Link)`
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;

  > img {
    height: 3rem;
  }
`;
