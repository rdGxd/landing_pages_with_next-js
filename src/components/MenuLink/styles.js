import Link from "next/link";
import styled, { css } from "styled-components";

export const Container = styled(Link)`
  ${({ theme }) => css`
    display: inline-flex;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primaryColor};
    position: relative;

    &::after {
      content: "";
      position: absolute;
      bottom: 0.76rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      transition: all 300ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }

    @media ${theme.media.lteMedium} {
      display: flex;
    }
  `}
`;
