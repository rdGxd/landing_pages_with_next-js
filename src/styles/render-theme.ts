import { render, RenderResult } from '@testing-library/react';
import { theme } from './theme';
import { ThemeProvider } from 'styled-components';

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};
