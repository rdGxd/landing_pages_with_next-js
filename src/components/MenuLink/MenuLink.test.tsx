import { screen } from "@testing-library/react";
import { MenuLink } from ".";
import { renderTheme } from "../../styles/render-theme";
// import { theme } from "../../styles/theme";

describe("<MenuLink />", () => {
  it("should render a link", () => {
    renderTheme(<MenuLink link="http://localhost">Children</MenuLink>);
    expect(screen.getByRole("link", { name: "Children" })).toHaveAttribute("target", "_self");
  });

  it("should render open in a new tab", () => {
    renderTheme(
      <MenuLink link="http://localhost" newTab={true}>
        Children
      </MenuLink>,
    );
    expect(screen.getByRole("link", { name: "Children" })).toHaveAttribute("target", "_blank");
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(
      <MenuLink link="http://localhost" newTab={false}>
        Children
      </MenuLink>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
