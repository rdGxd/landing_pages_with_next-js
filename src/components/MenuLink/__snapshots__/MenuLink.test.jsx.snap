// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match snapshot 1`] = `
.c0 {
  display: inline-flex;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0.76rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #DC143C;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width: 768px) {
  .c0 {
    display: flex;
  }
}

<a
  class="c0"
  href="http://localhost"
  target="_self"
>
  Children
</a>
`;
